projects/core/src/organization/store/actions/permission.action.ts

```diff
@@ -0,0 +1,150 @@
+import { Permission } from '../../../model/permission.model';
+import {
+  EntityFailAction,
+  EntityLoadAction,
+  EntitySuccessAction,
+} from '../../../state/utils/entity-loader/entity-loader.action';
+import { B2BSearchConfig } from '../../model/search-config';
+import { serializeB2BSearchConfig } from '../../utils/serializer';
+import { PERMISSION_ENTITIES, PERMISSION_LIST } from '../organization-state';
+import { ListModel } from '../../../model/misc.model';
+
+export const LOAD_PERMISSION = '[Permission] Load Permission Data';
+export const LOAD_PERMISSION_FAIL = '[Permission] Load Permission Data Fail';
+export const LOAD_PERMISSION_SUCCESS =
+  '[Permission] Load Permission Data Success';
+
+export const LOAD_PERMISSIONS = '[Permission] Load Permissions';
+export const LOAD_PERMISSIONS_FAIL = '[Permission] Load Permissions Fail';
+export const LOAD_PERMISSIONS_SUCCESS = '[Permission] Load Permissions Success';
+
+export const CREATE_PERMISSION = '[Permission] Create Permission';
+export const CREATE_PERMISSION_FAIL = '[Permission] Create Permission Fail';
+export const CREATE_PERMISSION_SUCCESS =
+  '[Permission] Create Permission Success';
+
+export const UPDATE_PERMISSION = '[Permission] Update Permission';
+export const UPDATE_PERMISSION_FAIL = '[Permission] Update Permission Fail';
+export const UPDATE_PERMISSION_SUCCESS =
+  '[Permission] Update Permission Success';
+
+export class LoadPermission extends EntityLoadAction {
+  readonly type = LOAD_PERMISSION;
+  constructor(public payload: { userId: string; permissionCode: string }) {
+    super(PERMISSION_ENTITIES, payload.permissionCode);
+  }
+}
+
+export class LoadPermissionFail extends EntityFailAction {
+  readonly type = LOAD_PERMISSION_FAIL;
+  constructor(permissionCode: string, public payload: any) {
```

Comment:
To make action more extendable it's better to have single parameter as an object. Then from payload you can extract whatever you need here.